### PR TITLE
tests: add user_presence tests

### PR DESCRIPTION
**Pull request by meenal06**
Signed-off-by: Meenal Trivedi <meenaltrivedi6102@gmail.com>

### CODE REVIEW

**Comment by meenal06** [comment]
Hi @Kegsay  , so to fix the 401, I created an unauthedClient and then made a user through that client. After than I created an authedClient with the userID of the user created from the unauthedClient, but I am getting error that the user was not found. How should I go ahead and approach the following tests 😕 ?

**Comment by meenal06** [comment]
Hi @Kegsay , I have fixed the tests and created authedClient as it was specified in the onboarding docs, Please do take a look. But I still have the previous doubt that why did my earlier method didn't work and what if we want to create a new user which we may need to use with an authedClient?

File: tests/apidoc_login_test.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ func TestLogin(t *testing.T) {
 	})
 }
 
-func createDummyUser(t *testing.T, unauthedClient *client.CSAPI, userID string) {
+func CreateDummyUser(t *testing.T, unauthedClient *client.CSAPI, userID string) {
```
Comment (Kegsay) [inline tests/apidoc_login_test.go:152]: Please keep this private. You don't need it for the new tests.

File: tests/apidoc_presence_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package tests
```
Comment (Kegsay) [inline tests/apidoc_presence_test.go:1]: Please add a blacklist in for Dendrite as it does not implement presence yet. E.g https://github.com/matrix-org/complement/blob/33f712f8b1c138210ee9514c32f383f10937bb7a/tests/user_query_keys_test.go#L1 - this comment is read by Go, and it has to be the first line in the file. Replace `synapse` with `dendrite`.

Comment (Kegsay) [reply tests/apidoc_presence_test.go:1]: You will also need to add this tag to the Github Actions file - https://github.com/matrix-org/complement/blob/09ac587f5a1c96f056b8723fc1316d1e816c228f/.github/workflows/ci.yaml#L29 - add in `dendrite_blacklist`.

Doing this should make CI pass.

**Review by Kegsay — CHANGES_REQUESTED** [review]

**Comment by meenal06** [comment]
Hi @Kegsay , please review I have made all the changes you requested 😄 !

**Review by Kegsay — APPROVED** [review]

**Pull request merged**